### ISSUE TITLE
EDD Purchase Link gets straight_to_gateway added when it shouldnt

### ISSUE THREAD

**Issue by mintplugins** [easydigitaldownloads/easy-digital-downloads#7192]
### Expected behavior
When I use the purchase_link shortcode with `direct="false"`, I expect to not be automatically redirected to the checkout when adding the item to my cart.

### Actual behavior
When I use the purchase_link shortcode with `direct="false"`, I am automatically redirected to the checkout when adding the item to my cart.

The false positive is happening here:
https://github.com/easydigitaldownloads/easy-digital-downloads/blob/b2ce5ed02d62737dd17ac665a71a493d9efe5898/includes/template-functions.php#L218

### Steps to reproduce the behavior
1) Put this shortcode on a page:
`[purchase_link id="YOUR_PRODUCT_ID_HERE" direct="false"]`
2) Click on the button, notice that you are redirected to the checkout automatically

### Information (if a specific version is affected):
PHP Version: 7.2

EDD Version (or branch): master

WordPress Version: 5.3

**Comment by cklosowski** [comment]
Direct means direct to gateway, so it should not go to the checkout.

**Comment by mintplugins** [comment]
Replicated on `master`. Confirmed fixed on `issue/7192`.

**Comment by cklosowski** [comment]
@mintplugins The `direct` is set to override a situation where you _do not_ want to go `Direct to gateway`. If `direct` is `false`, then I would want to go to the checkout.

if `direct` was `true` then I would want to go to the gateway directly (buy now).

If the "expected" is:

> When I use the purchase_link shortcode with direct="false", I expect to not be automatically redirected to the checkout when adding the item to my cart.

The `direct` is not weather you should automatically redirect to the cart, but if you are trying to force (or not force) a `Buy Now` behavior (from the docs)

> direct – If set to "true", it will use "Buy Now" mode, which bypasses the cart/checkout screen altogether. If set to "false", it will add the product to the cart.

**Comment by cklosowski** [comment]
The only way to affect 'automatic redirect to cart' is via the setting in EDD.

**Comment by mintplugins** [comment]
@cklosowski I suppose regardless of my expectations, if you add "direct="true" without a gateway that supports Buy Now, it takes you to the checkout automatically. 

That's the way it is in EDD core at this time. Is that a bug as well?

**Comment by mintplugins** [comment]
Oh actually, in order to replicate this, you need a gateway that does support Buy Now mode (ie Stripe/PayPal Standard), and taxes must be off. Then use the shortcode like this:

`[purchase_link id="YOUR_PRODUCT_ID_HERE" direct="false"]`

When clicking on it, you'll be auto-redirected to the cart.

**Comment by cklosowski** [comment]
> When clicking on it, you'll be auto-redirected to the cart.

That is expected. You have set the `direct` to be `false`, which means you've disabled `buy now` (direct to gateway)

**Comment by mintplugins** [comment]
Right, but why would I be directed to the checkout page instead of having it get added to the cart?

**Comment by cklosowski** [comment]
That would be a bug in the `redirect to cart`, not the `buy now/straight_to_gateway`. I need to setup a few tests and I'll figure out where the issue is at.

First we need to define that the `direct` has nothing to do with the action of redirecting to the cart, as it only applies to if we should trigger the `buy now/straight_to_gateway` actions, so we aren't messing with that logic.

**Comment by cklosowski** [comment]
I've updated the issue to focus solely on the concept of if the `direct` flag is affecting the `buy now` functionality.
